code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -481,14 +483,16 @@
 
 /obj/structure/mineral_door/resin/Destroy()
 	relativewall_neighbours()
-	var/turf/U = loc
+	var/area/area = get_area(src)
+	area?.current_resin_count--
 	spawn(0)
```

Comment:
```suggestion
	area?.current_resin_count--
	var/turf/base_turf = loc
	spawn(0)
```
You cannot do `get_step(loc, cardinal)` below, because it is inside of a `spawn()`, and by the time code gets to this point, the `..()` has already deleted the source object and there is no `loc` to use. That `var/turf/U = loc` was actually meaningful, without it you can be left with a freestanding door.
![image](https://github.com/user-attachments/assets/e20e199d-1cc8-4364-8842-9f50e4cfe249)